tests/components/recorder/test_migration_from_schema_32.py

```diff
@@ -844,3 +852,578 @@ def _fetch_migrated_events():
+        if row["start"] is not None:
+            row["start"] = process_timestamp(row["start"]).replace(tzinfo=None)
+        if row["last_reset"] is not None:
+            row["last_reset"] = process_timestamp(row["last_reset"]).replace(
+                tzinfo=None
+            )
+
+    assert final_result == [
+        {
+            "created": process_timestamp(one_year_ago).replace(tzinfo=None),
+            "created_ts": one_year_ago.timestamp(),
+            "id": 1,
+            "last_reset": process_timestamp(one_year_ago).replace(tzinfo=None),
+            "last_reset_ts": one_year_ago.timestamp(),
+            "max": None,
+            "mean": None,
+            "metadata_id": 1000,
+            "min": None,
+            "start": process_timestamp(one_year_ago).replace(tzinfo=None),
+            "start_ts": one_year_ago.timestamp(),
+            "state": 1.0,
+            "sum": None,
+        },
+        {
+            "created": None,
+            "created_ts": six_months_ago.timestamp(),
+            "id": 2,
+            "last_reset": None,
+            "last_reset_ts": six_months_ago.timestamp(),
+            "max": None,
+            "mean": None,
+            "metadata_id": 1000,
+            "min": None,
+            "start": None,
+            "start_ts": six_months_ago.timestamp(),
+            "state": 1.0,
+            "sum": None,
+        },
+        {
+            "created": process_timestamp(one_month_ago).replace(tzinfo=None),
+            "created_ts": one_month_ago.timestamp(),
+            "id": 3,
+            "last_reset": process_timestamp(one_month_ago).replace(tzinfo=None),
+            "last_reset_ts": one_month_ago.timestamp(),
+            "max": None,
+            "mean": None,
+            "metadata_id": 1000,
+            "min": None,
+            "start": process_timestamp(one_month_ago).replace(tzinfo=None),
+            "start_ts": one_month_ago.timestamp(),
+            "state": 1.0,
+            "sum": None,
+        },
+    ]
+
+
+async def test_stats_timestamp_with_one_by_one(
+    async_setup_recorder_instance: RecorderInstanceGenerator,
+    hass: HomeAssistant,
+) -> None:
+    """Test stats migration with one by one."""
+    instance = await async_setup_recorder_instance(hass)
+    await async_wait_recording_done(hass)
+    await async_attach_db_engine(hass)
+    importlib.import_module(SCHEMA_MODULE)
+    old_db_schema = sys.modules[SCHEMA_MODULE]
+    now = dt_util.utcnow()
+    one_year_ago = now - datetime.timedelta(days=365)
+    six_months_ago = now - datetime.timedelta(days=180)
+    one_month_ago = now - datetime.timedelta(days=30)
+
+    def _do_migration():
+        with patch.object(
+            migration,
+            "_migrate_statistics_columns_to_timestamp",
+            side_effect=IntegrityError("test", "test", "test"),
+        ):
+            migration._migrate_statistics_columns_to_timestamp_removing_duplicates(
```

Comment:
It would have been nicer to not patch here but including a full sqlite database for a test case is less maintainable